test/unit/integrations/paytm/paytm_module_test.rb

```diff
@@ -0,0 +1,152 @@
+require 'test_helper'
+require 'openssl'
+require 'base64'
+require 'digest'
+require 'securerandom'
+	
+	
+  
+
+class PaytmModuleTest < Test::Unit::TestCase
+  include OffsitePayments::Integrations
+  
+  def new_pg_encrypt(params)
+    if (params.class != Hash) || (params.keys == [])
+      return false
+    end
+    if !params.has_key?(:key)
+      return false
+    end
+    encrypted_data = Hash[]
+    key = params.delete(:key)
+    keys = params.keys
+    aes = OpenSSL::Cipher::Cipher.new("aes-128-cbc")
+    begin
+      keys.each do |k|
+        data = params[k]
+        aes.encrypt
+        aes.key = key
+        aes.iv = '@@@@&&&&####$$$$'
+        encrypted_k = aes.update(k.to_s) + aes.final
+        encrypted_k = Base64.encode64(encrypted_k.to_s)
+        aes.encrypt
+        aes.key = key
+        aes.iv = '@@@@&&&&####$$$$'
+        encrypted_data[encrypted_k] = aes.update(data.to_s) + aes.final
+        encrypted_data[encrypted_k] = Base64.encode64(encrypted_data[encrypted_k])
+      end
+    rescue Exception => e
+      return false
+    end
+    return encrypted_data
+  end
+  
+  
+  def new_pg_encrypt_variable(data, key)
+    aes = OpenSSL::Cipher::Cipher.new("aes-128-cbc")
+    aes.encrypt
+    aes.key = key
+    aes.iv = '@@@@&&&&####$$$$'
+    encrypted_data = nil
+    begin
+      encrypted_data = aes.update(data.to_s) + aes.final
+      encrypted_data = Base64.encode64(encrypted_data)
+    rescue Exception => e
+      return false
+    end
+    return encrypted_data
+  end 
+  
+
+  def new_pg_generate_salt(length)
+    salt = SecureRandom.urlsafe_base64(length*(3.0/4.0))
```

Comment:
Hi,

This is because we need 4 digit salt and urlsafe_base64 return n*4/3 digits so to get a salt of 4 digit formula is (4*(3/4)) 
for your reference kindly check http://ruby-doc.org/stdlib-1.9.3/libdoc/securerandom/rdoc/SecureRandom.html

Regards,